### ISSUE TITLE
Add utility function for retrieving packages with arbitrary relationships

### ISSUE THREAD

**Issue by ianling** [spdx/tools-golang#71]
https://github.com/spdx/tools-golang/blob/main/spdxlib/described_elements.go

This file contains functions that allow you to retrieve the IDs of the packages described by the document.

Would it be possible to get a similar function that allows us to retrieve IDs of packages that have a given relationship, e.g. DEPENDS_ON/DEPENDENCY_OF, rather than specifically DESCRIBES/DESCRIBED_BY?

Something like:

```
func GetPackagesByRelationship(doc *spdx.Document2_2, relationship string, ref spdx.ElementID) ([]spdx.ElementID, error) { ... }

usage:
GetPackagesByRelationship(doc, "DEPENDENCY_OF", packageA)
GetPackagesByRelationship(doc, "DEPENDS_ON", packageB)
GetPackagesByRelationship(doc, "DESCRIBED_BY", document)
```

**Comment by swinslow** [comment]
This is a great suggestion, thanks @ianling. Yes, I expect this would be very useful.

One consideration would be whether the search should be on just one side of the relationship (e.g. just RefA or just RefB), or should search both. Most of the [relationship types](https://spdx.github.io/spdx-spec/7-relationships-between-SPDX-elements/#71-relationship) are "directional" in that it's meaningful which is on the left vs. right side, but some I think probably are not (e.g. DYNAMIC_LINK, STATIC_LINK).

Perhaps part of the function call should be to specify whether to search just left side, just right side, or both?

Either way, yes, I'm a big fan of adding this. If you'd like to propose a PR, please feel free!

**Comment by swinslow** [comment]
Fixed in #74, thank you @ianling!
